controllers/storageconfig_controller.go

```diff
@@ -91,11 +94,23 @@ func (r *StorageSecretReconciler) reconcileSecret(secret *corev1.Secret,
 	if err != nil {
 		if apierrs.IsNotFound(err) {
 			log.Info("No data connections found in namespace ", secret.Namespace)
+			return nil
```

Comment:
This return changes quite a lot of the existing logic, it seems good to me (aka if there are no data connection it is fine to exit the method) but I find strange that the existing code is working: as far as I see the `get` for `foundStorageSecret` should fail and the flow should enter in the create branch.
Long story short I would expect the existing code to create an empty (?) secrets even if there is no data connection.
Am I missing something?
Btw I'm fine with the change, it is mainly to understand how it is changing the behaviour

Reply:
>Long story short I would expect the existing code to create an empty (?) secrets even if there is no data connection.
You are right! This is a bug and I found this during the test. Creating an empty `storage-config` does not make sense because a user can create it manually.  

With the return, it will not create an empty `storage-config` when dataconnection does not exist.

Reply:
Yes with the new return it is correct indeed, I was referring to the old / previous Logic 👍